src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/JsonClassDataContract.cs

```diff
@@ -87,31 +87,32 @@ internal JsonFormatClassWriterDelegate JsonFormatWriterDelegate
             }
         }
 
-        internal XmlDictionaryString[] MemberNames => _helper.MemberNames;
+        internal XmlDictionaryString[]? MemberNames => _helper.MemberNames;
 
         internal override string TypeName => _helper.TypeName;
 
         private ClassDataContract TraditionalClassDataContract => _helper.TraditionalClassDataContract;
 
-        public override object ReadJsonValueCore(XmlReaderDelegator jsonReader, XmlObjectSerializerReadContextComplexJson context)
+        public override object? ReadJsonValueCore(XmlReaderDelegator jsonReader, XmlObjectSerializerReadContextComplexJson? context)
         {
             jsonReader.Read();
             object o = JsonFormatReaderDelegate(jsonReader, context, XmlDictionaryString.Empty, MemberNames);
             jsonReader.ReadEndElement();
             return o;
         }
 
-        public override void WriteJsonValueCore(XmlWriterDelegator jsonWriter, object obj, XmlObjectSerializerWriteContextComplexJson context, RuntimeTypeHandle declaredTypeHandle)
+        public override void WriteJsonValueCore(XmlWriterDelegator jsonWriter, object obj, XmlObjectSerializerWriteContextComplexJson? context, RuntimeTypeHandle declaredTypeHandle)
```

Comment:
should obj be nullable?

Reply:
https://github.com/dotnet/runtime/pull/41476#discussion_r479391298